sklearn/cluster/tests/test_optics.py

```diff
@@ -224,7 +225,7 @@ def test_nowarn_if_metric_bool_data_bool():
     pairwise_metric = "rogerstanimoto"
     X = np.random.randint(2, size=(5, 2), dtype=bool)
 
-    with pytest.warns(None) as warn_record:
+    with warnings.catch_warnings(record=True) as warn_record:
         OPTICS(metric=pairwise_metric).fit(X)
         assert len(warn_record) == 0
```

Comment:
```suggestion
    with warnings.catch_warnings():
    	warnings.simplefilter("error", DataConversionWarning)
        OPTICS(metric=pairwise_metric).fit(X)
```

where `DataConversionWarning` is from `sklearn.exceptions`.

With this change, we can remove the `assert len(warn_record) == 0` below.